[menu-applet] a more intuitive way of keyboard-navigation in menu

NikoKrause opened the pull request:
When opening the menu you can see, that the "All Applications" button is highlighted:
![bildschirmfoto vom 2016-10-02 22-02-13](https://cloud.githubusercontent.com/assets/8415124/19023333/b19407c8-88ec-11e6-9188-e3a42a03858f.png)

Imagine the following two scenarios:
1. You would expect to get the next category by pressing the  down key on your keyboard, because the first category is already highligted. Wrong.
Instead you get the first application in the "All Applications" category.
2. You press the right key on your keyboard and expect to get the first application in the "All Applications" category. Wrong again. You get the first category "All Applications", which is already highlighted.

I think the suggested way is more intuitive, due to the fact, that the "All Applications" button is highlighted, when you open the menu.

The only thing I wasn't sure about was what the `this._scrollToButton(item_actor._delegate);` function is doing. Maybe somebody, who does know what the function is doing, could check that.

lestcape commented:
@NikoKrause  what is doing the _scrollToButton function, is move the scroll position of the applications, to a place where will  be show the selected application inside the visible viewport of the menu.

So, just take care to always pass as a parameter to this function the selected application  bottom and not the selected category bottom, because categories have not an scroll area.

NikoKrause commented:
So thanks to lestcape comment `scrollToButton` is now used only by the applicationsBox.
Also `previousVisibleIndes` is now only used by applicationsBox.

This fixes a weird behaviour, i.e. when you scrolled to the last application in "All Applications" category and went from there straight to the favoritesBox and scrolled the favoriteButtons.
First the applicationsBox jumped to the top, when you started scrolling in the favoritesBox and second the last application you visited in the applicationsBox was not remembered. 

I fixed that behavior and rebased the branch to one commit only.

JosephMcc commented:
Minor comment on the code. Remember conditionals should have a space between if  and ().

NikoKrause commented:
Thanks I'll change that

JosephMcc commented:
Are you active in our IRC channel? Maybe under a different name? Feel free to join anytime if you want to discuss changes or know what others are working on. It's #linuxmint-dev

NikoKrause commented:
I have no idea what an IRC channel is.

lestcape commented:
@NikoKrause https://community.linuxmint.com/tutorial/view/12

NikoKrause commented:
Ah thanks. I'll give it a try.

NikoKrause commented:
Added space between if and ().
